spring-cloud-task-app-starters-docs/src/main/asciidoc/overview.adoc

```diff
@@ -153,3 +153,184 @@ public class TimestampTaskApplication {
+`spring-cloud-task-app-generator` module and start editing as below.
+
+If you have a very basic task starter, all you have to do is to add the following to the list of apps need to be
+generated in the `generatedApps` tag of the configuration section of `spring-cloud-stream-app-maven` plugin.
+
+[source,xml]
+----
+<generatedApps>
+....
+    <foobar-task />
+....
+</generatedApps>
+----
+
+More information about the maven plugin used above can be found here:
+https://github.com/spring-cloud/spring-cloud-stream-app-maven-plugin
+
+If you did not follow the default convention expected by the plugin of where it is looking for the main configuration
+class, which is `org.springfamework.cloud.task.app.foobar.FoobarTaskConfiguration`, you can override that in
+the configuration for the plugin. For example, if your main configuration class is `foo.bar.SpecialFooBarTaskConfiguration.class`,
+this is how you can tell the plugin to override the default.
+
+[source,xml]
+----
+<foobar-task>
+    <autoConfigClass>foo.bar.SpecialFooBarTaskConfiguration.class</autoConfigClass>
+</foobar-task>
+----
+
+* Go to the root of the repository and execute the script: `./generateApps.sh`
+
+This would generate the foobar task app in a directory named `apps` at the root of the repository.
+If you want to change the location where the apps are generated, for instance /tmp/task-apps, you can do it in the
+configuration section of the plugin.
+
+[source,xml]
+----
+<configuration>
+    ...
+    <generatedProjectHome>/tmp/task-apps</generatedProjectHome>
+    ...
+</configuration
+----
+
+If you have an artifact that is only available through a private internal maven repository (may be an enterprise wide
+Nexus repo that you use globally across teams), and you need that for your app, you can define that as part of the maven
+plugin configuration.
+
+For example,
+
+[source,xml]
+----
+<configuration>
+...
+    <extraRepositories>
+        <repository>
+            <id>private-internal-nexus</id>
+            <url>.../</url>
+            <name>...</name>
+            <snapshotEnabled>...</snapshotEnabled>
+        </repository>
+    </extraRepositories>
+</configuration>
+----
+
+Then you can define this as part of your app tag:
+
+[source,xml]
+----
+<foobar-task>
+    <extraRepositories>
+        <private-internal-nexus />
+    </extraRepositories>
+</foobar-task>
+----
+
+* cd into the directory where you generated the apps (`apps` at the root of the repository by default, unless you changed
+it elsewhere as described above).
+
+Here you will see `foobar-task` along with all the other out of the box apps that is generated.
+
+If you only care about the foobar-task apps and nothing else, you can cd into that directory
+and import it directly into your IDE of choice. Each of them is a self contained spring boot application project.
+For all the generated apps, the parent is `spring-boot-starter-parent`. There is no way to change that around as ultimately these
```

Comment:
The reason why we can't change it is that it is a requirement of Spring Initialzr.